core/src/main/scala/cats/data/Chain.scala

```diff
@@ -611,28 +679,37 @@ sealed abstract class Chain[+A] extends ChainCompat[A] {
    * }}}
    */
   final def lengthCompare(len: Long): Int = {
-    import java.lang.Long
-    this match {
-      // `isEmpty` check should be faster than `== Chain.Empty`,
-      // but the compiler fails to prove that the match is still exhaustive.
-      case _ if isEmpty       => Long.compare(0L, len)
-      case Chain.Singleton(_) => Long.compare(1L, len)
-      case _ if len < 2       => 1 // the following cases should always have `length >= 2`
-      case Chain.Wrap(seq) =>
-        if (len > Int.MaxValue) -1 // `Seq#length` has `Int` type so cannot be `> Int.MaxValue`
-        else
-          seq.lengthCompare(len.toInt)
-      case _ => // should always be `Chain.Append` (i.e. `NonEmpty` with 2+ elements)
-        var sz = 2L
-        val it = new ChainIterator(this)
-        it.next()
-        it.next()
-        while (it.hasNext) {
-          if (sz == len) return 1
-          it.next()
-          sz += 1L
+    // This is an optimized (unboxed) implementation
+    // of the same code as foldLeft
+    @annotation.tailrec
+    def loop(head: Chain.NonEmpty[A], tail: List[Chain.NonEmpty[A]], acc: Long): Int =
+      if (acc < 0L) 1 // head is nonempty
```

Comment:
If I am not missing something, then `if (acc <= 0L) 1` should work as well.
Because we know that the head is not empty, so seems we can skip looking further if `acc == 0` is given.

Reply:
yep. Good call. Updated.